Action timer API (#2374)
DFHack/dfhack: docs/dev/Lua API.rst

  Returns a table of the cutoffs used by the above stress level functions.

> Action Timer API

Reviewer: You should mention somewhere (probably where you talk about overflow protection) that if an action is in progress, the minimum value that it will be set to is 1 so that the action is completed on the next tick. Similarly, action types that are currently at 0 can't be set to any other value since there is no associated action.

~~~~~~~~~~~~~~~~

This is an API to allow manipulation of unit action timers, to speed them up or slow them down.
All functions in this API have overflow/underflow protection when modifying action timers (the value will cap out).
``affectedActionType`` parameters are integers from the DF enum ``unit_action_type``. E.g. ``df.unit_action_type.Move``.
``affectedActionTypeGroup`` parameters are integers from the (custom) DF enum ``unit_action_type_group``. They are as follows:

  * ``All`` (does not include unknown unit action types)
  * ``Movement``

>   * ``MovementFeet`` (check if the unit is not on ground before using this one?)

Reviewer: Why is this a question? What happens when callers set this group (or All) to some value and the unit is not on the ground?

Author: I'm not sure how to word it honestly.

  Returns a table of the cutoffs used by the above stress level functions.

Action Timer API
~~~~~~~~~~~~~~~~

This is an API to allow manipulation of unit action timers, to speed them up or slow them down.
All functions in this API have overflow/underflow protection when modifying action timers (the value will cap out).
Actions with a timer of 0 (or less) will not be modified as they are completed (or invalid in the case of negatives).
Timers will be capped to go no lower than 1.
``affectedActionType`` parameters are integers from the DF enum ``unit_action_type``. E.g. ``df.unit_action_type.Move``.
``affectedActionTypeGroup`` parameters are integers from the (custom) DF enum ``unit_action_type_group``. They are as follows:

  * ``All`` (does not include unknown unit action types)
  * ``Movement``

>   * ``MovementFeet`` (intended to only be used when the unit is standing, such as with pegasus boots from the modding guide)

Reviewer: I guess my question for this is: why does it matter if the dwarf is on the ground? This is just a timer, isn't it? Why is it bad to modify it in since conditions? Fastdwarf doesn't bother to check if the dwarf is standing before setting all timers, which includes the MovementFeet timers. Is that bad?

Author: No the idea is that if you want something to only work when you are standing, like the pegasus boots, because they work with your feet, you would check if you are standing and if you are then use the MovementFeet category. But fastdwarf has nothing to do with that case, as it wants all timers to be faster, so there's nothing wrong there.
Sorry if it's still not clear!

>   * ``MovementFeet`` (intended to only be used when the unit is standing, such as with pegasus boots from the modding guide)

Reviewer: ```suggestion
  * ``MovementFeet`` (for walking speed, such as with pegasus boots from the `modding-guide`)
```

Reviewer: is this ok?
